### ISSUE TITLE
keyword-spacing: Do not treat "this" as keyword

### ISSUE THREAD

**Issue by Alaneor** [eslint/eslint#5043]
Current implementation of the [`keyword-spacing`](https://github.com/eslint/eslint/blob/master/docs/rules/keyword-spacing.md) rule treats `this` as a keyword, but it's not a keyword - it's a variable.

I propose to remove `this` from being checked by this rule.

ESLint: 2.0.0-beta.2

**Comment by eslintbot** [comment]
@Alaneor Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Could you provide a snippet of code to reproduce this issue with an error output please?

**Comment by Alaneor** [comment]
Apologies for not including that right away.

**Error:**

`1:15  warning  Expected space(s) before "this"  keyword-spacing`

**Offending code:**

```js
return `${this.prefix}`
```

**Comment by michaelficarra** [comment]
It is certainly a keyword and not a variable.

**Comment by Alaneor** [comment]
[I concur](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this).

However, in the context of this rule, I still find it a little weird to be included here. Most of the time, we treat `this` as a variable holding the execution context, compared to "things" like `if`, `else`, `var`, `function`, `return` etc. which are used for control flow, declarations or other effects.

I am genuinely curious to know what code would trigger this rule by improper spacing around `this` (except for my current issue with template strings). All the examples I could come up with are already covered in other rules and are mentioned in the examples detailing how this rule does not conflict with them. Thanks!

**Comment by mysticatea** [comment]
Thank you for this issue!

[`this` is a keyword](http://www.ecma-international.org/ecma-262/6.0/#sec-keywords), but this is a bug.
`keyword-spacing` is designed to ignore spacing between `{` and a keyword.

I will fix this.

**Comment by Alaneor** [comment]
Thank you for fixing this issue!
